src/app/code/community/Zendesk/Zendesk/controllers/Adminhtml/ZendeskController.php

```diff
@@ -129,11 +154,25 @@ public function logoutAction()
         $adminSession->getCookie()->delete($adminSession->getSessionName());
         $adminSession->addSuccess(Mage::helper('adminhtml')->__('You have logged out.'));
 
-        $this->_redirect('*');
+        $this->_redirect('adminhtml/zendesk/*');
     }
 
     public function createAction()
     {
+        try {
+            $domain = Mage::getStoreConfig('zendesk/general/domain');
+            if(!$domain) {
+                Mage::getSingleton('adminhtml/session')->addError(Mage::helper('zendesk')->__('Please set up Zendesk connection.'));
+                $this->_redirect('adminhtml/zendesk/index');
+                return;
+            }
+        } catch( Exception $ex ) {
```

Comment:
When would this exception be thrown? Seems like we're being too defensive here and also throwing exactly the same error wouldn't help with debugging.